processor/transformprocessor/internal/traces/traces.go

```diff
@@ -121,325 +135,325 @@ func newPathGetSetter(path []common.Field) (getSetter, error) {
 		},
-		setter: func(ctx spanTransformContext, val interface{}) {
+		setter: func(ctx common.TransformContext, val interface{}) {
 			if attrs, ok := val.(pdata.Map); ok {
-				ctx.resource.Attributes().Clear()
-				attrs.CopyTo(ctx.resource.Attributes())
+				ctx.GetResource().Attributes().Clear()
+				attrs.CopyTo(ctx.GetResource().Attributes())
 			}
 		},
 	}
 }
 
 func accessResourceAttributesKey(mapKey *string) pathGetSetter {
 	return pathGetSetter{
-		getter: func(ctx spanTransformContext) interface{} {
-			return getAttr(ctx.resource.Attributes(), *mapKey)
+		getter: func(ctx common.TransformContext) interface{} {
+			return getAttr(ctx.GetResource().Attributes(), *mapKey)
 		},
-		setter: func(ctx spanTransformContext, val interface{}) {
-			setAttr(ctx.resource.Attributes(), *mapKey, val)
+		setter: func(ctx common.TransformContext, val interface{}) {
+			setAttr(ctx.GetResource().Attributes(), *mapKey, val)
 		},
 	}
 }
 
-func accessInstrumentationLibrary() pathGetSetter {
+func accessInstrumentationScope() pathGetSetter {
 	return pathGetSetter{
-		getter: func(ctx spanTransformContext) interface{} {
-			return ctx.il
+		getter: func(ctx common.TransformContext) interface{} {
+			return ctx.GetInstrumentationScope()
 		},
-		setter: func(ctx spanTransformContext, val interface{}) {
+		setter: func(ctx common.TransformContext, val interface{}) {
 			if newIl, ok := val.(pdata.InstrumentationScope); ok {
-				newIl.CopyTo(ctx.il)
+				newIl.CopyTo(ctx.GetInstrumentationScope())
 			}
 		},
 	}
 }
 
-func accessInstrumentationLibraryName() pathGetSetter {
+func accessInstrumentationScopeName() pathGetSetter {
 	return pathGetSetter{
-		getter: func(ctx spanTransformContext) interface{} {
-			return ctx.il.Name()
+		getter: func(ctx common.TransformContext) interface{} {
+			return ctx.GetInstrumentationScope().Name()
 		},
-		setter: func(ctx spanTransformContext, val interface{}) {
+		setter: func(ctx common.TransformContext, val interface{}) {
 			if str, ok := val.(string); ok {
-				ctx.il.SetName(str)
+				ctx.GetInstrumentationScope().SetName(str)
 			}
 		},
 	}
 }
 
-func accessInstrumentationLibraryVersion() pathGetSetter {
+func accessInstrumentationScopeVersion() pathGetSetter {
 	return pathGetSetter{
-		getter: func(ctx spanTransformContext) interface{} {
-			return ctx.il.Version()
+		getter: func(ctx common.TransformContext) interface{} {
+			return ctx.GetInstrumentationScope().Version()
 		},
-		setter: func(ctx spanTransformContext, val interface{}) {
+		setter: func(ctx common.TransformContext, val interface{}) {
 			if str, ok := val.(string); ok {
-				ctx.il.SetVersion(str)
+				ctx.GetInstrumentationScope().SetVersion(str)
 			}
 		},
 	}
 }
 
 func accessTraceID() pathGetSetter {
 	return pathGetSetter{
-		getter: func(ctx spanTransformContext) interface{} {
-			return ctx.span.TraceID()
+		getter: func(ctx common.TransformContext) interface{} {
+			return ctx.GetItem().(pdata.Span).TraceID()
```

Comment:
Does something need to be done to ensure that this cannot panic if `GetItem()` returns something other than a `pdata.Span`?